sds/src/secondary_validation/dutch_passport_checksum.rs

```diff
@@ -0,0 +1,101 @@
+use crate::secondary_validation::Validator;
+
+pub struct DutchPassportChecksum;
+
+const DUTCH_PASSPORT_LENGTH: usize = 9;
+
+impl Validator for DutchPassportChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        /*
+         * Dutch passport numbers are 9 characters long:
+         * - First two characters are letters from A-N or P-Z (excluding O)
+         * - Next 6 characters can be either digits or letters from A-N or P-Z (excluding O)
+         * - Last character is a checksum digit calculated using the ICAO 9303 algorithm
+         */
+        if regex_match.len() != DUTCH_PASSPORT_LENGTH {
+            return false;
+        }
+
+        // Check first two characters are valid letters (A-N or P-Z)
+        if !regex_match[..2].chars().all(|c| {
+            let upper = c.to_ascii_uppercase();
+            upper.is_ascii_alphabetic() && upper != 'O'
+        }) {
+            return false;
+        }
+
+        // Check next 6 characters are valid (digits or A-N or P-Z)
+        if !regex_match[2..8].chars().all(|c| {
+            if c.is_ascii_digit() {
+                true
+            } else {
+                let upper = c.to_ascii_uppercase();
+                upper.is_ascii_alphabetic() && upper != 'O'
+            }
+        }) {
+            return false;
+        }
+
+        // Last character must be a digit (checksum)
+        if !regex_match.chars().last().unwrap().is_ascii_digit() {
```

Comment:
Similarly, only the `.len()` of the string was checked, which is the number of _bytes_, but this `.unwrap()` is asserting that the number of _characters_ which will panic if there is a multi-byte char anywhere in the match.